network-api/networkapi/buyersguide/management/commands/migrate_products.py

```diff
@@ -192,21 +193,24 @@ def handle(self, *args, **options):
                 new_product_page.get_or_create_votes()
                 # Use .to_dict() to pull out the old aggregated votes
                 product_dict = product.to_dict()
-                votes = product_dict['votes']['creepiness']['vote_breakdown']
-                values = [x for x in votes.values()]
-                new_product_page.set_votes(values)
-                new_product_page.current_vote_count = product_dict['votes']['total']
-
-            # Save and/or publish the final page.
+                if hasattr(product_dict, 'votes'):
+                    votes = product_dict['votes']['creepiness']['vote_breakdown']
+                    values = [x for x in votes.values()]
+                    product_total = product_dict['votes']['total']
+                else:
+                    # Default vote "bin"
+                    values = [0, 0, 0, 0, 0]
+                    product_total = 0
+
+            new_product_page.votes.set_votes(values)
+            new_product_page.current_vote_count = product_total
             new_product_page.save()
-            if not product.draft:
-                new_product_page.save_revision().publish()
-            else:
-                new_product_page.save_revision()
 
             # Always good to fresh from db when using Django Treebeard.
             buyersguide_page.refresh_from_db()
 
+        time.sleep(1)
```

Comment:
Gives the DB a quick second to catch up.. just in case. I ran into a couple issues where my db was still updating locally and couldnt find ProductPage's afterwards. This fixed it.

Reply:
That should be impossible: Django db operations are synchronous...

Reply:
I doubt this would be a problem for production. Locally I think my database simply couldnt update fast enough — possibly a locked table?

Regardless, I figured a 1 second pause in a temporary migration script wouldnt be the worst thing to have.